src/KMonad/Model/Action.hs

```diff
@@ -93,8 +97,10 @@ makeClassy ''Trigger
 
 -- | ADT signalling where to install a hook
 data HookLocation
-  = InputHook  -- ^ Install the hook immediately after receiving a 'KeyEvent'
-  | OutputHook -- ^ Install the hook just before emitting a 'KeyEvent'
+  = InputHookPrio   -- ^ Install the hook immediately after receiving a 'KeyEvent'
+                    -- and before the sluice
+  | InputHook       -- ^ Installthe hook after the sluice
```

Comment:
I would have added a parmater to the `InputHook` to differentiate whether before or after the sluice.
This would make the `within` function more sensical as `within OutputHook` is a bit wierd.